testing.rst

```diff
@@ -590,6 +590,12 @@ container is a service that can be get via the normal container::
         // ...
     }
 
+.. tip::
```

Comment:
```suggestion
.. caution::
```
?

Reply:
Sorry, I merged directly whilst it was still a question.